test/custom_class.ll

```diff
@@ -0,0 +1,75 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc
+; ModuleID = 'custom_class.ll'
+source_filename = "/localdisk2/icl/fadeeval/sycl_workspace/repo2/custom_class.cpp"
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-linux-sycldevice"
+
+%class._ZTS11CustomClass.CustomClass = type { i32 }
+%"class._ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEEUlvE_.anon" = type { i8 }
+
+$"_ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEE10SYCLCustom" = comdat any
+
+@_ZN11CustomClassC1Ei = dso_local unnamed_addr alias void (%class._ZTS11CustomClass.CustomClass addrspace(4)*, i32), void (%class._ZTS11CustomClass.CustomClass addrspace(4)*, i32)* @_ZN11CustomClassC2Ei
+
+; Function Attrs: norecurse
+define weak_odr dso_local spir_kernel void @"_ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEE10SYCLCustom"() #0 comdat !kernel_arg_addr_space !4 !kernel_arg_access_qual !4 !kernel_arg_type !4 !kernel_arg_base_type !4 !kernel_arg_type_qual !4 {
```

Comment:
Could this function be also deleted? Or without it this test doesn't crash?

Reply:
The bug is because of this function so I can't delete it.

Reply:
Ok then.

Reply:
I found that the crash can be reproduced if you delete this function and remove "internal" specifier from definition of "_ZZZ4mainENK3$_0clERN2cl4sycl7handlerEENKUlvE_clEv" function.
However, I don't know if this is a valid modification (removing of internal linkage) for current test case.